typing/typecore.ml

```diff
@@ -6345,6 +6344,63 @@ let type_clash_of_trace trace =
     | _ -> None
   ))
 
+(** More precise denomination for type errors. Used by messages:
+
+    - "This <denom> ..."
+    - "The <denom> 'foo' ..." *)
+let pp_exp_denom ppf exp =
+  let f = fprintf ppf "%s" in
+  let desc_expression = fprintf ppf "%a expression" Style.inline_code in
+  match exp.pexp_desc with
+  | Pexp_constant _ -> f "constant"
+  | Pexp_ident _ -> f "value"
+  | Pexp_construct _ | Pexp_variant _ -> f "constructor"
+  | Pexp_field _ -> f "field access"
+  | Pexp_send _ -> f "method call"
+  | Pexp_while _ -> desc_expression "while"
+  | Pexp_for _ -> desc_expression "for"
+  | Pexp_ifthenelse _ -> desc_expression "if-then-else"
+  | Pexp_match _ -> desc_expression "match"
+  | Pexp_try _ -> desc_expression "try-with"
+  | _ -> f "expression"
+
+let report_general_this_exp denom ppf = fprintf ppf "This %t" denom
+
+let pp_inline_exp =
+  let pp_exp ppf exp =
+    match exp.pexp_desc with
+    | Pexp_constant { pconst_desc = Pconst_string (s, _, None); _ } ->
+        (* Avoid adding quotes around a quoted string constant but make sure to
+           use consistent inline code style. *)
+        Format.pp_print_string ppf (String.escaped s)
```

Comment:
This is only an issue with dumb terminal and in the testsuite, isn't it? Moreover using `String.escaped` does not mesh well with non-ascii contents. I would propose to skip this step.

Reply:
You are right, making the strings nominal is problematic. I removed it.